torch/_refs/__init__.py

```diff
@@ -6335,9 +6335,8 @@ def _infer_scalar_type(obj):
 # xref: recursive_store in torch/csrc/utils/tensor_new.cpp
 def _recursive_build(scalarType: torch.dtype, obj: TensorOrNumberLikeType):
     if isinstance(obj, Tensor) and obj.ndim <= 1:
-        obj = obj.item()
-        # fall through into next case
-    if isinstance(obj, Number):
+        return obj.to(dtype=scalarType, device="cpu", copy=True)
```

Comment:
Is the scalar_tensor() constructor setting some special flag that this would not?
Also do we need a .squeeze() to collapse all the dimensions?

Reply:
It's not, scalar_tensor is basically the same as tensor (but it can only make 0d tensors). And yes, need the squeeze.